mobilecoind/clients/python/cli/README.md

```diff
@@ -0,0 +1,114 @@
+
+Note that this command creates a monitor for the provided or generated Master Key that will remain active until removed.
+
+Help:
+```
+usage: get_public_address.py [-h] [-k KEY] [-s [SUBADDRESS]]
+
+Displays public address information for a provided master key, or for a random master key if no key is provided.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -k KEY, --key KEY     account master key
+  -s [SUBADDRESS], --subaddress [SUBADDRESS]
+                        (optional) subaddress
+```
+
+Example:
+```
+mobilecoin@node1:~/mobilecoin/mobilecoind/clients/python/cli$ python3 get_public_address.py
+
+    Master Key:       2a4976758b6d8d687ba6aea1e934d8b6511a264e6b979ed74c58534e7b575a5b
+    Subaddress Index: 0
+    Address Code:     d6vRUHZW4dRQqmzmmoLtdejkgoAS3jicYKryHzgAGmUjAPyqo1LjEugi9tpBVYfWxywhxXncYvkVBboERH87voJAZQeiTnkuPretVnnrSb
+    Address URL:
+
+```
+
+### Checking your Balance
+
+Because the MobileCoin ledger is encrypted, you must process every transaction with your Master Key in order to check your balance. This CLI tool creates an account monitor, waits for it to process all blocks in the ledger, and then displays the total unspent balance controlled by the provided Master Key.
+
+Note that this command creates a monitor for the provided Master Key that will remain active until removed.
+
+Help:
+```
+usage: check_balance.py [-h] -k KEY [-s [SUBADDRESS]]
+                        [--first-block [FIRST_BLOCK]]
+
+Processes all downloaded blocks to display current balance information for a provided master key.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -k KEY, --key KEY     account master key
+  -s [SUBADDRESS], --subaddress [SUBADDRESS]
+                        (optional) subaddress
+  --first-block [FIRST_BLOCK]
+                        (optional) first ledger block to scan
+
+```
+
+Example:
+```
+mobilecoin@node1:~/mobilecoin/mobilecoind/clients/python/cli$ python3 check_balance.py -k=<master key (64 hex characters)>
+
+    Master Key:       <master key (64 hex characters)>
+    Subaddress Index: 0
+    Address Code:     7BeDc5jpZu72AuNavumc8qo8CRJijtQ7QJXyPo9dpnqULaPhe6GdaDNF7cjxkTrDfTcfMgWVgDzKzbvTTwp32KQ78qpx7bUnPYxAgy92caJ
+    Address URL:
+    Balance:          998401053000000000 pMOB
+                      998.40 kMOB
+
+```
+
+### Sending a Payment
+
+Send a payment.
+
+Help:
+```
+usage: send_payment.py [-h] --sender SENDER --recipient RECIPIENT [-v VALUE]
+                       [--sender-subaddress [SENDER_SUBADDRESS]]
+                       [--recipient-subaddress [RECIPIENT_SUBADDRESS]]
+
+Send all available funds (default) or VALUE pMOB from a sender's Master Key to a recipient, specified by Master Key or Address Code.
```

Comment:
I'm not sure that draining the account should be the default. Would maybe suggest erroring if no value is provided.

Reply:
I find myself wanting to fold one wallet into another fairly often, usually because I'm done with a key or because I moved funds to a key and then a script failed before the key was assigned etc... I can make the user add "--all" instead of having this as a default.

Reply:
I think `--all` makes sense - the negative experience of accidentally sending all your tokens seems like it should have a protection against it.